File: app/src/ultimate/java/com/clockmods/widget/update/WidgetRefreshPolicy.java
package com.clockmods.widget.update;
public final class WidgetRefreshPolicy {
    private WidgetRefreshPolicy() { }
    public static boolean shouldSchedule(int weatherCount) { return weatherCount>0; }
    public static boolean shouldRetry(boolean configured,boolean transientFailure,int attempt) { return configured && transientFailure && attempt<2; }
}

File: app/src/ultimate/java/com/clockmods/widget/update/WidgetRefreshPolicy.kt
package com.clockmods.widget.update

object WidgetRefreshPolicy {
    @JvmStatic fun shouldSchedule(weatherCount: Int): Boolean = weatherCount > 0

    @JvmStatic fun shouldRetry(configured: Boolean, transientFailure: Boolean, attempt: Int): Boolean =
        configured && transientFailure && attempt < 2
}
